src/dlgProfilePreferences.cpp

```diff
@@ -1158,33 +1158,37 @@ void dlgProfilePreferences::initWithHost(Host* pHost)
     connect(doubleSpinBox_networkPacketTimeout, qOverload<double>(&QDoubleSpinBox::valueChanged), this, &dlgProfilePreferences::slot_setPostingTimeout);
 
     //Shortcuts tab
-    QMapIterator<QString, QKeySequence*> i(pHost->profileShortcuts);
+    auto shortcutKeys = mudlet::self()->mShortcutsManager->iterator();
     int shortcutsRow = 0;
-    while (i.hasNext()) {
-        i.next();
-        QKeySequence* sequence = new QKeySequence(*i.value());
-        currentShortcuts.insert(i.key(), sequence);
+    while (shortcutKeys.hasNext()) {
+        auto key = shortcutKeys.next();
+        QKeySequence* sequence = new QKeySequence(*pHost->profileShortcuts.value(key));
+        currentShortcuts.insert(key, sequence);
         auto sequenceEdit = new QKeySequenceEdit(*sequence);
 
-        gridLayout_groupBox_shortcuts->addWidget(new QLabel(i.key()), floor(shortcutsRow / 2), (shortcutsRow % 2) * 2 + 1);
+        gridLayout_groupBox_shortcuts->addWidget(new QLabel(key), floor(shortcutsRow / 2), (shortcutsRow % 2) * 2 + 1);
         gridLayout_groupBox_shortcuts->addWidget(sequenceEdit, floor(shortcutsRow / 2), (shortcutsRow % 2) * 2 + 2);
         shortcutsRow++;
         connect(sequenceEdit, &QKeySequenceEdit::editingFinished, this, [=]() {
-            QKeySequence newSequence;
+            QKeySequence* newSequence = nullptr;
             if (sequenceEdit->keySequence().isEmpty()) {
-                newSequence = *sequence;
+                newSequence = sequence;
             } else if (sequenceEdit->keySequence().matches(QKeySequence(Qt::Key_Escape))) {
-                newSequence = QKeySequence();
+                newSequence = new QKeySequence();
             } else {
-                newSequence = sequenceEdit->keySequence();
+                newSequence = new QKeySequence(sequenceEdit->keySequence());
             }
             sequenceEdit->blockSignals(true);
-            sequenceEdit->setKeySequence(newSequence);
+            sequenceEdit->setKeySequence(*newSequence);
             sequenceEdit->blockSignals(false);
-            sequence->swap(newSequence);
+            sequence->swap(*newSequence);
+            delete newSequence;
         });
         connect(this, &dlgProfilePreferences::signal_resetMainWindowShortcutsToDefaults, sequenceEdit, [=]() {
-            sequenceEdit->setKeySequence(*mudlet::self()->mShortcutsManager->getDefault(i.key()));
+            sequenceEdit->setKeySequence(*mudlet::self()->mShortcutsManager->getDefault(key));
```

Comment:
it looks advisable (from the previous one) to prevent the `sequenceEdit` object from being triggered whilst the key-sequence is being manipulated!
```suggestion
            sequenceEdit->blockSignals(true);
            sequenceEdit->setKeySequence(*mudlet::self()->mShortcutsManager->getDefault(key));
            sequenceEdit->blockSignals(false);
```

Reply:
Actually other way around. Previously used `keySeuqenceChanged` which cause looping. I forgot that I can safely remove previous guards.